### ISSUE TITLE
UX/Accessibility: Content warnings and "Show more" buttons are not obvious to new users

### ISSUE THREAD

**Issue by DarkFox** [mastodon/mastodon#30724]
### Steps to reproduce the problem

1. Create a post with a content warning.
2. Share the link to this post with a person who is not familiar with Mastodon.
3. Observe their confusion about the content of the post.

### Expected behaviour

The new user should be able to easily identify and understand the "Show more" button and content warnings

### Actual behaviour

The new user thought that the content warning was the entire post, and didn't even register the "Show more" button

### Detailed description

When sharing posts with content warnings on Mastodon, new users who are not familiar with the platform may find the "Show more" button and the content warning itself to be unclear. This issue is exacerbated by poor color contrast, making the button hard to see or appear disabled, especially for people with poor vision.

For example, when I shared a post with a content warning about mental health and medication, the recipient only saw the text "MH+, MH-, medication, neurodivergence" and did not understand that this was a content warning, and not the post itself, and that the "Show more" button would show the post itself. This confusion highlights a need for better UX design to make content warnings and "Show more" buttons more intuitive and visually distinct.

I've attached a screenshot of the current UI -- using the "Contrast loss" filter in Firefox's dev tools -- for reference. As seen in the screenshot, the "Show more" button blends in with the background, making it not immediately obvious that it is interactive and will reveal the hidden content when clicked.

![image](https://github.com/mastodon/mastodon/assets/28210/a11ea624-3b39-41d2-8289-45af7a0224ba)

I experienced this issue on a glitch-soc instance, but I can see that the UI is the exact same on plain Mastodon.

### Mastodon instance

tech.lgbt

### Mastodon version

v4.3.0-alpha.4+glitch

### Browser name and version

Firefox 126.0.1 (64-bit)

### Operating system

Windows 10 Home 22H2, 19045.4474

**Comment by steinareliassen** [comment]
That it needs to be more clear that "show more" is a button to push to expand text, is one aspect.

That the text there is written as a content warning, is another. As it is now, it can feel like it is just a "intro paragraph" with "click to view the rest".

**Comment by vmstan** [comment]
Potentially addressed by #31365

**Comment by DarkFox** [comment]
> Potentially addressed by #31365

I believe it just might be.

The symbolism is a lot better, with the warning strips along the sides, and "Show anyway" makes it clear the message is being hidden due to the CW listed above. Some might still not understand that they're content warnings, but I think most would get the idea.
